pkg/controller/installation/installer.go

```diff
@@ -154,38 +154,77 @@ func (i *Installer) patchDeployment(
 	}
 	d.Spec.Template.SetLabels(podTemplateLabels)
 
-	return d
+	return d, nil
 }
 
 func (i *Installer) patchService(
 	s *corev1.Service,
 	labelsToInject map[string]string,
 	ownerReference *metav1.OwnerReference,
-) runtime.Object {
-	appName, ok := labelsToInject[shipper.AppLabel]
-	if !ok {
-		// If we reach this point, it is fine to panic() since there's nothing
-		// much to do if we don't have the application label.
-		panic(fmt.Sprintf("Programmer error, label %q should always be present", shipper.AppLabel))
+) (runtime.Object, error) {
+
+	requiredLabels := []string{shipper.AppLabel}
+	for _, label := range requiredLabels {
+		if _, ok := labelsToInject[label]; !ok {
+			// If we reach this point, it is fine to panic() since there's nothing
```

Comment:
I think this comment is now outdated.

Reply:
Fixed in https://github.com/bookingcom/shipper/pull/43/commits/bbb4bac9ac90257461bb74a137f4fbb29c2d5a8b